### PR TITLE
Feat/15622577 permissions filters

### PR DESCRIPTION
**Pull request by KornelKwiatkowski**
## Description

- added new permission filters

**New stuff** ✨

image

(Resolves | Contributes to) #31415

### CODE REVIEW

**Review by krawczykkatarzyna — APPROVED** [review]

File: src/components/v5/common/Filter/types.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ export type FilterOptionProps = {
   id: NestedFilterOption;
   title: Message;
   icon?: JSX.Element;
+  children?: FilterOptionProps[];
```
Comment (CzarekDryl) [inline src/components/v5/common/Filter/types.ts:39]: I don't know if children is a good name, maybe something like `nestedOptions` or `subOptions`.

File: src/components/v5/shared/SubNavigationItem/partials/NestedOptions.tsx [hunk 2]
```diff
@@ -17,9 +20,37 @@ const NestedOptions: FC<NestedOptionsProps> = ({
...
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    if (hasNestedOptions) {
+      if (isChecked) {
+        setCheckedParent(null);
+        return;
+      }
+
+      setCheckedParent(id);
+      return;
```
Comment (TomaszChmielPagepro) [inline src/components/v5/shared/SubNavigationItem/partials/NestedOptions.tsx:48]: `setCheckedParent(isChecked ? null : id)`

File: src/context/FilterContext.tsx [hunk 3]
```diff
@@ -171,25 +170,34 @@ export const FilterContextProvider: FC<PropsWithChildren> = ({ children }) => {
     setSelectedFilterCount((prevState) => prevState - removedFilters);
   }, []);
 
-  const getFilterDomainIds = useCallback(
-    () =>
+  const getFilterDomainIds = useCallback(() => {
+    return (
```
Comment (TomaszChmielPagepro) [inline src/context/FilterContext.tsx:174]: `return` is not needed

File: src/components/v5/common/Filter/types.ts (hunk 1 above)
Comment (KornelKwiatkowski) [reply src/components/v5/common/Filter/types.ts]: changed

**Review by CzarekDryl — APPROVED** [review]

File: src/components/v5/shared/SubNavigationItem/partials/NestedOptions.tsx (hunk 2 above)
Comment (KornelKwiatkowski) [reply src/components/v5/shared/SubNavigationItem/partials/NestedOptions.tsx:50]: done

File: src/context/FilterContext.tsx (hunk 3 above)
Comment (KornelKwiatkowski) [reply src/context/FilterContext.tsx:174]: done

**Review by TomaszChmielPagepro — APPROVED** [review]

**Pull request merged**